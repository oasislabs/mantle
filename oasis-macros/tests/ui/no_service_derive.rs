#![feature(proc_macro_hygiene)]
#[oasis_std::service]
mod service {
    pub struct Counter(u32);

    impl Counter {
        pub fn new(ctx: &Context) -> Result<Self> {
            Ok(Self(42))
        }
    }
}

fn main() {}
