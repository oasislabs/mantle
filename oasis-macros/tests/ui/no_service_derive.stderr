error: Service definition must contain a #[derive(Service)] struct.
 --> $DIR/no_service_derive.rs:2:1
  |
2 | #[oasis_std::service]
  | ^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

